src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -88,53 +91,69 @@ type ContextMenuData = ContextMenuProps["ContextMenu"] & {
-
-  const repluggedGroup = <MenuGroup />;
-  repluggedGroup.props.id = "replugged";
-  repluggedGroup.props.children = [];
-
-  // Add in the new menu items right above the DevMode Copy ID
-  // If the user doesn't have DevMode enabled, the new items will be at the bottom
-  menu.children.splice(-1, 0, repluggedGroup);
-
-  menuItems[navId].forEach((item) => {
-    try {
-      const res = makeItem(item.getItem(data, menu)) as ContextItem & { props: { id?: string } };
-
-      if (res?.props) {
-        // add in unique ids
-        res.props.id = `${res.props.id || "repluggedItem"}-${Math.random()
-          .toString(36)
-          .substring(2)}`;
+    ([menu]: [ContextMenuData], res) => {
+      const ContextMenu = res as React.FunctionComponent<ContextMenuData>;
+      const { navId } = menu;
+      const { MenuGroup } = RepluggedComponents as Record<string, React.ComponentType>;
+
+      // No items to insert
+      // Or MenuGroup Component is not available
+      if (!menuItems[navId] || !MenuGroup) {
+        return <ContextMenu {...menu} />;
       }
 
-      menu.children.at(item.sectionId)?.props?.children?.splice(item.indexInSection, 0, res);
-    } catch (err) {
-      logger.error("Error while running GetContextItem function", err, item.getItem);
-    }
-  });
+      // The data as passed as Arguments from the calling function, so we just grab what we want from it
+      const data = menu.data[0];
+
+      const repluggedGroup = <MenuGroup />;
+      repluggedGroup.props.id = "replugged";
+      repluggedGroup.props.children = [];
+
+      // Add in the new menu items right above the DevMode Copy ID
+      // If the user doesn't have DevMode enabled, the new items will be at the bottom
+      menu.children.splice(-1, 0, repluggedGroup);
+      menuItems[navId].forEach((item) => {
+        try {
+          const itemRet = makeItem(item.getItem(data, menu)) as ContextItem & {
+            props: { id?: string };
+          };
+
+          if (itemRet?.props) {
+            // add in unique ids
+            itemRet.props.id = `${itemRet.props.id || "repluggedItem"}-${Math.random()
+              .toString(36)
+              .substring(2)}`;
+          }
+
+          if (!menu.children.at(item.sectionId)?.props?.plugged) {
+            // menu props can stay same while having different children
+            // so we check children props if its plugged or not
+            menu.children
+              .at(item.sectionId)
+              ?.props.children?.splice(item.indexInSection, 0, itemRet);
+            if (menu.children.at(item.sectionId)?.props)
+              menu.children.at(item.sectionId)!.props.plugged = true;
+          }
+        } catch (err) {
+          logger.error("Error while running GetContextItem function", err, item.getItem);
+        }
+      });
+
+      return <ContextMenu {...menu} />;
+    },
+  );
+}
+export function start(): void {
+  injector = new Injector();
+  void injectMenu();
```

Comment:
You can move the implementation in start, a dedicated function is not needed in this case.